ui-speedspacechart/src/components/helpers/drawElements/tickX.ts

```diff
@@ -17,65 +17,83 @@ export const drawTickX = ({ ctx, width, height, store }: DrawFunctionParams) =>
   ctx.font = 'normal 12px IBM Plex Sans';
   ctx.fillStyle = 'rgb(182, 179, 175)';
 
-  const { maxPosition, RoundMaxPosition, intermediateTicksPosition } = maxPositionValues(store);
+  const maxPosition = maxPositionValue(store);
+  const windowLength = maxPosition / store.ratioX;
+
+  // Define the tick scale and the principle tick frequency given the window length
+  let tickScale = 0.1;
+  let principleTickFrequency = 5;
+  if (windowLength >= 200) {
+    tickScale = Math.floor(windowLength / 200) * 5;
+    principleTickFrequency = 2;
+  } else if (windowLength >= 50) {
+    tickScale = 2;
+    principleTickFrequency = 5;
+  } else if (windowLength >= 20) {
+    tickScale = 1;
+    principleTickFrequency = 5;
```

Comment:
Do we need to re-set `principleTickFrequency` to 5 even though it's initialized with this value on l.25 ?

Reply:
This is not required. However, this value is very much tied to `tickScale`, more than to the condition. That's why I like them next to each other. If you modify one of the values, it's very likely that the other will also be modified.

In rust I would have written it like that:

```rust
let (tickScale, principleTickFrequency) = if windowLength >= 200 {
      ((windowLenght / 200.0).floor() * 5.0, 2)
  } else if windowLenght >= 50 {
      (2.0, 5)
  } else if windowLenght >= 20 {
      (1.0, 5)
  } else {
      (0.1, 5)
  };
```

Do you think there's a way to write this in TS that matches the code above?

Reply:
exactly like you did, you could also declare an object or a tupple but it seems over-engineered to me

in this specific case, the repetition seems useless for the logic itself, and even if your point is understandable it's not a concept usually used in TS

Reply:
to get the best of two worlds:

```js
let tickScale: number;
let principleTickFrequency: number;
if (windowLength >= 200) {
    tickScale = Math.floor(windowLength / 200) * 5;
    principleTickFrequency = 2;
} else if (windowLength >= 50) {
    tickScale = 2;
    principleTickFrequency = 5;
} else if (windowLength >= 20) {
    tickScale = 1;
    principleTickFrequency = 5;
} else {
    tickScale = 0.1;
    principleTickFrequency = 5;
}
```